File: js_checkio_solutions/Electronic Station/ascending_list.js
#!/usr/bin/env checkio --domain=js run ascending-list
"use strict";
// Determine whether the sequence of elements items is  ascending so that its each element is  strictly larger   than (and not merely equal to) the element that precedes it.
// 
// Input:Iterable with ints.
// 
// Output:Bool.
// 
// The mission was taken from Python CCPS 109 Fall 2018. It is taught for Ryerson Chang School of Continuing Education byIlkka Kokkarinen
// 
// 
// END_DESC
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var assert_1 = __importDefault(require("assert"));
function isAscending(values) {
    // 1. test emtpy
    if (Array.isArray(values) && ((values.length === 0) || (values.length === 1))) {
        return true;
    }
    var lastValue = values[0];
    for (var i = 1; i < values.length; i++) {
        var currentVal = values[i];
        if (lastValue >= currentVal) {
            return false;
        }
        lastValue = currentVal;
    }
    return true;
}
console.log('Example:');
console.log(isAscending([-5, 10, 99, 123456]));
// These "asserts" are used for self-checking
assert_1.default.equal(isAscending([-5, 10, 99, 123456]), true);
assert_1.default.equal(isAscending([99]), true);
assert_1.default.equal(isAscending([4, 5, 6, 7, 3, 7, 9]), false);
assert_1.default.equal(isAscending([]), true);
assert_1.default.equal(isAscending([1, 1, 1, 1]), false);
console.log("Coding complete? Click 'Check' to earn cool rewards!");

File: py_checkio_solutions/Electronic Station/ascending_list.py
#!/usr/bin/env checkio --domain=py run ascending-list

# Determine whether the sequence of elements items is  ascending so that its each element is  strictly larger   than (and not merely equal to) the element that precedes it.
# 
# Input:Iterable with ints.
# 
# Output:Bool.
# 
# The mission was taken from Python CCPS 109 Fall 2018. It is taught for Ryerson Chang School of Continuing Education byIlkka Kokkarinen
# 
# 
# END_DESC

from typing import Iterable
def is_ascending(items: Iterable[int]) -> bool:
    
    n = len(items)
    if n == 0 or n == 1:
        return True

    # Setting iterable items    
    iter_item = iter(items)

    prev_x = next(iter_item)

    while True:
        try:
            current_x = next(iter_item)
            if prev_x < current_x:
                prev_x = current_x
            else:
                return False
        except StopIteration:
            break

    return True


if __name__ == '__main__':
    print("Example:")
    print(is_ascending([-5, 10, 99, 123456]))
    
    # These "asserts" are used for self-checking and not for an auto-testing
    assert is_ascending([-5, 10, 99, 123456]) == True
    assert is_ascending([99]) == True
    assert is_ascending([4, 5, 6, 7, 3, 7, 9]) == False
    assert is_ascending([]) == True
    assert is_ascending([1, 1, 1, 1]) == False
    print("Coding complete? Click 'Check' to earn cool rewards!")